process/headerCheck/headerSignatureVerify.go

```diff
@@ -278,18 +285,26 @@ func (hsv *HeaderSigVerifier) getLeader(header data.HeaderHandler) (crypto.Publi
 	return hsv.keyGen.PublicKeyFromByteArray(leaderPubKeyValidator.PubKey())
 }
 
-func (hsv *HeaderSigVerifier) copyHeaderWithoutSig(header data.HeaderHandler) data.HeaderHandler {
+func (hsv *HeaderSigVerifier) copyHeaderWithoutSig(header data.HeaderHandler) (data.HeaderHandler, error) {
 	headerCopy := header.ShallowClone()
-	headerCopy.SetSignature(nil)
-	headerCopy.SetPubKeysBitmap(nil)
-	headerCopy.SetLeaderSignature(nil)
+	err := headerCopy.SetSignature(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = headerCopy.SetPubKeysBitmap(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	err = headerCopy.SetLeaderSignature(nil)
 
-	return headerCopy
+	return headerCopy, err
 }
 
-func (hsv *HeaderSigVerifier) copyHeaderWithoutLeaderSig(header data.HeaderHandler) data.HeaderHandler {
+func (hsv *HeaderSigVerifier) copyHeaderWithoutLeaderSig(header data.HeaderHandler) (data.HeaderHandler, error) {
 	headerCopy := header.ShallowClone()
-	headerCopy.SetLeaderSignature(nil)
+	err := headerCopy.SetLeaderSignature(nil)
 
```

Comment:
The same here -> return nil, err and return headerCopy, nil otherwise

Reply:
done